packages/panzer/disc-fe/src/lof/Panzer_BlockedEpetraLinearObjFactory_impl.hpp

```diff
@@ -537,26 +542,32 @@ buildDomainContainer() const
   using Teuchos::rcp;
 
   // if a "single field" DOFManager is used, return a single RO_GED
-  if(!colDOFManagerContainer_->containsBlockedDOFManager()) {
-    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> ged = rcp(new EpetraVector_ReadOnly_GlobalEvaluationData);
-    ged->initialize(getGhostedColImport(0),getGhostedColMap(0),getColMap(0));
-
+  if (!colDOFManagerContainer_->containsBlockedDOFManager())
+  {
+    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> ged =
+      rcp(new EpetraVector_ReadOnly_GlobalEvaluationData);
+    ged->initialize(getGhostedColImport2(0), getGhostedColMap2(0),
+      getColMap(0));
     return ged;
   }
 
   std::vector<RCP<ReadOnlyVector_GlobalEvaluationData> > gedBlocks;
-  for(int i=0;i<getBlockColCount();i++) {
-    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> vec_ged = rcp(new EpetraVector_ReadOnly_GlobalEvaluationData);
-    vec_ged->initialize(getGhostedColImport(i),getGhostedColMap(i),getColMap(i));
-
+  for (int i(0); i < getBlockColCount(); ++i)
+  {
+    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> vec_ged =
+      rcp(new EpetraVector_ReadOnly_GlobalEvaluationData);
+    vec_ged->initialize(getGhostedColImport2(i), getGhostedColMap2(i),
+      getColMap(i));
```

Comment:
`buildDomainContainer` will now use version 2 of `getGhostedColImport` and `getGhostedColMap`.